api-tester/src/main/java/com/revenuecat/apitester/kotlin/UpgradeInfoAPI.kt

```diff
@@ -0,0 +1,23 @@
+package com.revenuecat.apitester.kotlin
+
+import com.android.billingclient.api.BillingFlowParams
+import com.revenuecat.purchases.UpgradeInfo
+
+@Suppress("unused", "UNUSED_VARIABLE", "RemoveExplicitTypeArguments")
+private class UpgradeInfoAPI {
+    fun check(upgradeInfo: UpgradeInfo) {
+        with(upgradeInfo) {
+            val oldProductId: String = oldProductId
+            @BillingFlowParams.ProrationMode val prorationMode: Int? = prorationMode
+
+            val constructedUpgradeInfo =
+                UpgradeInfo(
+                    oldProductId,
+                    prorationMode
+                )
+
+            val constructedUpgradeInfoProductIdOnly = UpgradeInfo(oldProductId)
+            val constructedUpgradeInfoNullProrationMode = UpgradeInfo(oldProductId, null)
```

Comment:
I think this one would be caught by `constructedUpgradeInfo` since `prorationMode` is of type `Int?` so I think we can probably remove it.